src/main/java/uk/gov/ons/ctp/response/action/message/CaseNotificationReceiver.java

```diff
@@ -3,66 +3,41 @@
   @Autowired private CaseNotificationService caseNotificationService;
 
-  private static ProjectSubscriptionName getPubSubSubscription(
-      String project, String subscriptionId) {
-    log.with(subscriptionId, project)
-        .info("creating pubsub subscription for case notification with subscriptionId in project");
-    ProjectSubscriptionName subscriptionName = ProjectSubscriptionName.of(project, subscriptionId);
-    return subscriptionName;
-  }
-
   /**
-   * This method provides an Event Listener subscription for the action case notification.
+   * This is the message activator for the PubSub subscription which will receive the action case
+   * notification
    *
-   * @returns void
-   * @throws IOException
+   * @param message
    */
-  @EventListener(ApplicationReadyEvent.class)
-  public void caseNotificationSubscription() throws IOException {
-    // Instantiate an asynchronous message receiver.
-    MessageReceiver receiver =
-        (PubsubMessage message, AckReplyConsumer consumer) -> {
-          // Handle incoming message, then ack the received message.
-          log.with(message.getMessageId()).info("Receiving message ID from PubSub");
-          log.with(message.getData().toString()).debug("Receiving data from PubSub ");
-          try {
-            CaseNotification caseNotificationReceiver =
-                objectMapper.readValue(message.getData().toStringUtf8(), CaseNotification.class);
-            caseNotificationService.acceptNotification(caseNotificationReceiver);
-            consumer.ack();
-          } catch (final IOException | CTPException e) {
-            log.with(e)
-                .error(
-                    "Something went wrong while processing message received from PubSub for case notification");
-            consumer.nack();
-          }
-        };
-    Subscriber subscriber = pubSub.getActionCaseNotificationSubscriber(receiver);
-    // Start the subscriber.
-    subscriber.startAsync().awaitRunning();
-    log.with(pubSub.getActionCaseNotificationSubscriptionName().toString())
-        .info("Listening for action case notification messages on PubSub-subscription id");
+  @ServiceActivator(inputChannel = "actionCaseNotificationChannel")
+  public void messageReceiver(Message message) {
+    String payload = new String((byte[]) message.getPayload());
+    log.with("payload", payload).info("New request to register action case Notification");
+    try {
+      log.info("Mapping payload to CaseNotification object");
+      CaseNotification caseNotificationReceiver =
+          objectMapper.readValue(payload, CaseNotification.class);
+      log.info("Mapping successful, accepting action case notification");
+      caseNotificationService.acceptNotification(caseNotificationReceiver);
+    } catch (final IOException | CTPException e) {
+      log.with(e)
+          .error(
+              "Something went wrong while processing message received from PubSub for action case notification");
```

Comment:
will this stop it nacking the message? 

I think spring relies on exceptions to know whether to nack or ack the messages

Reply:
From here: https://cloud.spring.io/spring-cloud-gcp/multi/multi__spring_integration.html
```text
Automatic acking (AckMode.AUTO)

A message is acked with GCP Pub/Sub if the adapter sent it to the channel and no exceptions were thrown. If a RuntimeException is thrown while the message is processed, then the message is nacked.
```

Reply:
you are right! changed it to Manual mode.

Reply:
updated